threestudio/data/image.py

```diff
@@ -178,6 +179,9 @@ def collate(self, batch) -> Dict[str, Any]:
 
         return batch
 
+    def update_step(self, epoch: int, global_step: int, on_load_weights: bool = False):
```

Comment:
We actually don't need to do this. If SingleImageIterableDataset is Updateable, all its attributes that are Updateable (like `self.random_pose_generator` here) will be updated automatically. Could you please check this?

Reply:
I checked it, it didn't work. I had to insert an update_step that called self.random_pose_generator's update_step.